app.go

```diff
@@ -191,6 +191,24 @@ func (a *App) useShortOptionHandling() bool {
 func (a *App) Run(arguments []string) (err error) {
 	a.Setup()
 
+	commandNamesWithoutDuplicates := make(map[string]bool)
+	for _, c := range a.Commands {
+		_, cExists := commandNamesWithoutDuplicates[c.Name]
+		if cExists {
+			return fmt.Errorf("list of commands contains duplicate Name: %s", c.Name)
```

Comment:
⭐⭐ _(seriously consider)_

```suggestion
			return fmt.Errorf("Your app contains multiple commands with the Name %q. Having multiple commands with the same name results in ambiguous behavior, so please make sure each command in your app has a unique name.", c.Name)
```